Build failure: pkgsCross.wasi32.llvmPackages.libcxx is broken

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

On staging, aarch64-darwin:

```
nix build --file . pkgsCross.wasi32.llvmPackages.libcxx
```

### Can Hydra reproduce this build failure?

No, Hydra cannot reproduce this build failure.

### Link to Hydra build job

_No response_

### Relevant log output

```console
#
# Collect all the predefined macros, except for compiler version macros
# which we don't need to track here.
#
wasm32-unknown-wasi-clang -O2 -DNDEBUG --target=wasm32-wasi -fno-trapping-math -Wall -Wextra -Wno-null-pointer-arithmetic -Wno-unused-parameter -Wno-sign-compare -Wno-unused-variable -Wno-unused-function -Wno-ignored-attribut
es -Wno-missing-braces -Wno-ignored-pragmas -Wno-unused-but-set-variable -Wno-unknown-warning-option -Wno-unterminated-string-initialization -mthread-model single -Ilibc-bottom-half/cloudlibc/src -isystem "/nix/store/rx7r6mwd
2ndzhs6wm7splx4whi8js80k-wasilibc-static-wasm32-unknown-wasi-27-dev/include" "/nix/store/2csn56l6kpm3ccbljvd8g9w9c7d9f8s1-wasilibc-static-wasm32-unknown-wasi-27-share/share/include-all.c" \
    -isystem /nix/store/rx7r6mwd2ndzhs6wm7splx4whi8js80k-wasilibc-static-wasm32-unknown-wasi-27-dev/include \
    -std=gnu17 \
    -E -dM -Wno-\#warnings \
    -D_ALL_SOURCE \
    -U__llvm__ \
    -U__clang__ \
    -U__clang_major__ \
    -U__clang_minor__ \
    -U__clang_patchlevel__ \
    -U__clang_version__ \
    -U__clang_literal_encoding__ \
    -U__clang_wide_literal_encoding__ \
    -U__wasm_extended_const__ \
    -U__wasm_mutable_globals__ \
    -U__wasm_sign_ext__ \
    -U__wasm_multivalue__ \
    -U__wasm_reference_types__ \
    -U__wasm_nontrapping_fptoint__ \
    -U__wasm_bulk_memory__ \
    -U__wasm_bulk_memory_opt__ \
    -U__GNUC__ \
    -U__GNUC_MINOR__ \
    -U__GNUC_PATCHLEVEL__ \
    -U__VERSION__ \
    -U__NO_MATH_ERRNO__ \
    -U__BITINT_MAXWIDTH__ \
    -U__FLT_EVAL_METHOD__ -Wno-builtin-macro-redefined \
    | sed -e 's/__[[:upper:][:digit:]]*_ATOMIC_\([[:upper:][:digit:]_]*\)_LOCK_FREE/__compiler_ATOMIC_\1_LOCK_FREE/' \
    | sed -e 's/__GNUC_VA_LIST $/__GNUC_VA_LIST 1/' \
    | grep -v '^#define __\(BOOL\|INT_\(LEAST\|FAST\)\(8\|16\|32\|64\)\|INT\|LONG\|LLONG\|SHRT\)_WIDTH__' \
    | grep -v '^#define __GCC_HAVE_SYNC_COMPARE_AND_SWAP_\(1\|2\|4\|8\)' \
    | grep -v '^#define __FPCLASS_' \
    | grep -v '^#define __FLT128_' \
    | grep -v '^#define __MEMORY_SCOPE_' \
    | grep -v '^#define __GCC_\(CON\|DE\)STRUCTIVE_SIZE' \
    | grep -v '^#define __STDC_EMBED_' \
    | grep -v '^#define __\(DBL\|FLT\|LDBL\)_NORM_MAX__' \
    | grep -v '^#define NDEBUG' \
    | grep -v '^#define __OPTIMIZE__' \
    | grep -v '^#define assert' \
    | grep -v '^#define __NO_INLINE__' \
    | grep -v '^#define __U\?INT.*_C(' \
    > "/nix/store/2csn56l6kpm3ccbljvd8g9w9c7d9f8s1-wasilibc-static-wasm32-unknown-wasi-27-share/share/predefined-macros.txt"
Warning: supplying the --target wasm32-wasi != wasm32-unknown-wasi argument to a nix-wrapped compiler may not work correctly - cc-wrapper is currently not designed with multi-target compilers in mind. You may want to use an u
n-wrapped compiler instead.
clang: warning: argument unused during compilation: '-rtlib=compiler-rt' [-Wunused-command-line-argument]
# Check that the computed metadata matches the expected metadata.
# This ignores whitespace because on Windows the output has CRLF line endings.
diff -wur "expected/wasm32-wasip1" "/nix/store/2csn56l6kpm3ccbljvd8g9w9c7d9f8s1-wasilibc-static-wasm32-unknown-wasi-27-share/share"
diff -wur expected/wasm32-wasip1/predefined-macros.txt /nix/store/2csn56l6kpm3ccbljvd8g9w9c7d9f8s1-wasilibc-static-wasm32-unknown-wasi-27-share/share/predefined-macros.txt
--- expected/wasm32-wasip1/predefined-macros.txt        1970-01-01 00:00:01.000000000 +0000
+++ /nix/store/2csn56l6kpm3ccbljvd8g9w9c7d9f8s1-wasilibc-static-wasm32-unknown-wasi-27-share/share/predefined-macros.txt        2025-10-22 23:08:44.075294376 +0000
@@ -2159,6 +2159,7 @@
 #define _IONBF 2
 #define _ISO646_H 
 #define _LANGINFO_H 
+#define _LIBCPP_HARDENING_MODE _LIBCPP_HARDENING_MODE_EXTENSIVE
 #define _LIBGEN_H 
 #define _LIMITS_H 
 #define _LOCALE_H 
make: *** [Makefile:958: check-symbols] Error 1
```

### Additional context

This blocks Firefox build since it's a dependency via wasiSysRoot in pkgs/build-support/build-mozilla-mach/default.nix

Firefox is currently broken for Darwin on master so I cannot demonstrate Hydra failure.

### System metadata

```
 - system: `"aarch64-darwin"`
 - host os: `Darwin 24.6.0, macOS 15.6.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.28.5`
 - nixpkgs: `/nix/store/lqh77v3g5x9ais0wxagnibrw3sjw2019-source`
```

### Notify maintainers



---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/wa/wasilibc/package.nix
+++ b/pkgs/by-name/wa/wasilibc/package.nix
@@ -19,6 +19,12 @@ stdenvNoLibc.mkDerivation (finalAttrs: {
     fetchSubmodules = true;
   };
 
+  # These flags break pkgsCross.wasi32.llvmPackages.libcxx
+  hardeningDisable = [
+    "libcxxhardeningfast"
+    "libcxxhardeningextensive"
+  ];
+
   outputs = [
     "out"
     "dev"

